packages/projection-typeorm/src/TypeormStabilityWindowBuffer.ts

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable brace-style */
+import { BlockDataEntity } from './entity';
+import { Cardano, ChainSyncEventType } from '@cardano-sdk/core';
+import { FindOptionsSelect, LessThan, QueryRunner } from 'typeorm';
+import { Logger } from 'ts-log';
+import { Observable, ReplaySubject, concatMap, from, map } from 'rxjs';
+import {
+  Operators,
+  RollBackwardEvent,
+  RollForwardEvent,
+  StabilityWindowBuffer,
+  UnifiedProjectorOperator,
+  WithBlock
+} from '@cardano-sdk/projection';
+import { WithLogger, contextLogger } from '@cardano-sdk/util';
+import { WithTypeormContext } from './types';
+
+const blockDataSelect: FindOptionsSelect<BlockDataEntity> = {
+  block: {
+    slot: true
+  },
+  // Using 'transformers' breaks the types.
+  // Types seem to expect model to have fields that match database types:
+  // If it's an 'object', it will recursively apply FindOptionsSelect.
+  data: true as any,
+  id: true
+};
+
+export interface TypeormStabilityWindowBufferProps {
+  /**
+   * 100 by default
+   */
+  compactBufferEveryNBlocks?: number;
+}
+
+export class TypeormStabilityWindowBuffer
+  implements StabilityWindowBuffer<Operators.WithNetworkInfo & WithTypeormContext>
+{
+  #tail: Cardano.Block | 'origin';
+  readonly #logger: Logger;
+  readonly #compactEvery: number;
+  readonly #tip$ = new ReplaySubject<Cardano.Block | 'origin'>(1);
+  readonly #tail$ = new ReplaySubject<Cardano.Block | 'origin'>(1);
```

Comment:
This could be a `BehaviorSubject` and get rid of #tail property.
```diff
-  readonly #tail$ = new ReplaySubject<Cardano.Block | 'origin'>(1);
+  readonly #tail$ = new BehaviorSubject<Cardano.Block | 'origin'>('origin');
+  // use this.#tail$.value instead of this.#tail

Reply:
Hm not really, BehaviorSubject requires initial value and we don't want to emit `origin`. Having it with `| null` could work but then we'd need to expose it as `pipe(filter(isNotNil))` which I'm not sure is any better than existing implementation